File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.ts
function minOperations(nums: number[], x: number): number {
    const n = nums.length;
    const total = nums.reduce((a, b) => a + b, 0);
    const target = total - x;

    if (target < 0) return -1;
    if (target === 0) return n;

    let left = 0;
    let sum = 0;
    let longest = -1;

    for (let right = 0; right < n; right++) {
        sum += nums[right];
        while (left <= right && sum > target) {
            sum -= nums[left++];
        }
        if (sum === target) {
            longest = Math.max(longest, right - left + 1);
        }
    }

    return longest === -1 ? -1 : n - longest;
}

File: 1658-minimum-operations-to-reduce-x-to-zero/1658-minimum-operations-to-reduce-x-to-zero.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>, x: i32) -> i32 {
        let n = nums.len();
        let total: i32 = nums.iter().sum();
        let target = total - x;

        if target < 0 {
            return -1;
        }
        if target == 0 {
            return n as i32;
        }

        let mut left = 0;
        let mut sum = 0;
        let mut longest = -1;

        for right in 0..n {
            sum += nums[right];
            while left <= right && sum > target {
                sum -= nums[left];
                left += 1;
            }
            if sum == target {
                longest = longest.max((right - left + 1) as i32);
            }
        }

        if longest == -1 {
            -1
        } else {
            n as i32 - longest
        }
    }
}